enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -648,7 +648,10 @@ classDiagram
 - What is the performance impact when pods are gated and inspected over many
   namespaces? This needs to be qualified. Mitigations have already been
   discussed in the other sections, but we expect others to address at
-  implementation and performance evaluation time.
+  implementation and performance evaluation time. Leader election time
+  should also be considered when pods are gated and the primary controller is
+  down. This would add seconds worth of delay to the scheduling. Is there
+  scope to minimize this?
```

Comment:
If you can use `ReleaseOnCancel` then that will help to prevent long periods, but, to use `ReleaseOnCancel` you need to be sure you're shutting down correctly all reconciles with context management etc

Reply:
Hi @JoelSpeed , are you concerned about the possibility of some pods being scheduled without the nodeAffinity patch due to the missing schedulingGate, about the potential delay that scheduling gated pods might experience during the scheduling process, or both?

Reply:
i think the concern here is more about a race with both controllers being active if  there are issues in termination: https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go#L153

Reply:
What's the critical path in the case of the PodPlacementController?

We are configuring the PodPlacementController as active-passive for HA because it's challenging to assign shards of pods to (active-active) replicas of the controller reliably.

Using releaseOnCancel, if multiple reconciliations run concurrently on the same pod object, the Kubernetes optimistic concurrency principle ensures that only one process will succeed while the others will fail. 

The pods scheduling process will continue succeeding though.

Reply:
I'm concerned that a Pod may get the gate added, and then not be removed for an extended period (compared to normal operation), because the controller is shutting down and needs time to hand over to the next pod. If you use ReleaseOnCancel, the lease handover can be quite quick, without it, it can take upwards of a minute with normal OpenShift leader configuration

Reply:
Added a point discussing the releaseOnCancel option as a mitigation to the performance-related risks.